pointcloud/basic_octree.go

```diff
@@ -146,15 +146,44 @@ func (octree *BasicOctree) Pose() spatialmath.Pose {
 }
 
 // AlmostEqual compares the octree with another geometry and checks if they are equivalent.
-// TODO (RSDK-3743): Implement BasicOctree Geometry functions.
 func (octree *BasicOctree) AlmostEqual(geom spatialmath.Geometry) bool {
-	return false
+	otherOctree, ok := geom.(*BasicOctree)
+	if !ok {
+		return false
+	}
+	if octree.size != otherOctree.size {
```

Comment:
[minor] Potentially unneeded as you iterate through all points but for completeness we should check all metadata associated with the octree not just size.

So similar checks for label, meta, sideLength and center

Reply:
And technically this should occur for every subnode too so a recursive structure is advisable

Reply:
I would argue that two octrees are "almost equal" when they contain identical datapoints, without regard to their internal structures. So I think the sideLength, center, and meta checks are unnecessary.

Reply:
Alternately, I don't actually need this function for this PR; can just remove it.

Reply:
Up to you! I just wanted to confirm how we are defining AlmostEquals. For example, if we don't check the label then you may have two maps that are equal but one has labels and the other doesn't. Is that ok? We don't really have a use case yet so whatever we decide is probably are fine.

Reply:
Can you add to the comment to make it clear how we are defining AlmostEqual? i.e. same base size and contains same point locations.

Reply:
Other geometry `AlmostEqual` methods do not check labels either, so current convention is that two differently-labeled octrees created from the same pointcloud will indeed be AlmostEqual.